src/automata/SubscriptionRequester.cpp

```diff
@@ -7,6 +7,22 @@
 
 namespace reactivesocket {
 
+void SubscriptionRequesterBase::sendRequestFrame(
+    FrameFlags flags,
+    size_t initialN,
+    Payload&& request) {
+  Frame_REQUEST_SUB frame(
+      streamId_,
+      flags,
+      static_cast<uint32_t>(initialN),
+      FrameMetadata::empty(),
+      std::move(request));
+  // We must inform ConsumerMixin about an implicit allowance we have
+  // requested from the remote end.
+  addImplicitAllowance(initialN);
```

Comment:
One last nit: do you want to put this line to the onNext method, right before you call sendRequestFrame?
I don't see why moving this line up one spot would be a problem.